core/comp_ui.py

```diff
@@ -44,12 +45,14 @@ def _PromptLen(prompt_str):
     elif c == '\x02':
       escaped = False
     elif not escaped:
-      display_str += c
+      display_str = display_str + c
```

Comment:
hm maybe add a comment `# mycpp rewrite of +=`  here

I did that in a few other places where I didn't quite like how it turned out